propagator/ottrace/test/opentelemetry/propagator/ottrace/text_map_injector_test.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+module OpenTelemetry
+  module Propagator
+    module OTTrace
+      class TextMapInjector
+        TRACE_ID_64_BIT_WIDTH = 64 / 4
+
+        # Returns a new TextMapInjector that injects context using the specified setter
+        #
+        # @param [optional Setter] default_setter The default setter used to
+        #   write context into a carrier during inject. Defaults to a
+        #   {OpenTelemetry::Context:Propagation::TextMapSetter} instance.
+        # @return [TextMapInjector]
+        def initialize(default_setter = Context::Propagation.text_map_setter)
+          @default_setter = default_setter
+        end
+
+        # @param [Context] context The active Context.
+        # @param [optional Setter] setter If the optional setter is provided, it
+        #   will be used to write context into the carrier, otherwise the default
+        #   setter will be used.
+        # @return [Object] the carrier with context injected
+        def inject(carrier, context, setter = default_setter)
+          span_context = Trace.current_span(context).context
+          return carrier unless span_context.valid?
+
+          setter.set(carrier, TRACE_ID_HEADER, span_context.hex_trace_id[-TRACE_ID_64_BIT_WIDTH, TRACE_ID_64_BIT_WIDTH])
```

Comment:
@ericmustin I went with this implementation, which mirrors the JS implementation. Do you have any concerns about this?

Reply:
lgtm! I think either [-16,16] or [16,16] [like jaeger does](https://github.com/open-telemetry/opentelemetry-ruby/blob/a44514d2d85f53d377fb59a5a042c6946c085fff/exporter/jaeger/lib/opentelemetry/exporter/jaeger/encoder.rb#L115) is fine if i understand the constraints here (we don't expect >32character length string)

Reply:
Done!